### ISSUE TITLE
Interface injection cannot transform non-mod libraries

### ISSUE THREAD

**Issue by zml2008** [FabricMC/fabric-loom#646]
(yes, this is likely a bit of a challenge to implement even if there aren't dependency cycle issues from resolving interface injection data, but thought i might as well toss the issue up)

Now that Mixins targeting library classes are officially supported in loader 0.14, it would be nice to be able to inject interfaces into those libraries. 

In `adventure-platform-fabric` we implement our Adventure interfaces onto both game classes, which interface injection works great for, and onto Mojang-provided library classes [like Authlib's GameProfile](https://github.com/KyoriPowered/adventure-platform-fabric/pull/30/files#diff-e6d217ba0f07e84cf0220df007dfd74537d1fe097d57dd717f5b629b65dec99aR25), which we can't directly inject interfaces into. Supporting interface injection on these library classes would improve usability for end users.

I suspect allowing mods to inject interfaces into other mods' classes would open up even more thorny issues, so it's not something I'm asking for in this issue, but it's a feature worth considering while anyone who feels motivated to tackle this issue is looking in the area.

**Comment by modmuss50** [comment]
Yes, this would be nice to have. But for game libs its especially hard (more work, more to go wrong) as they dont go through the same remapping stage mods do. Worth leaving this issue open, but idk when it will get done.

**Comment by modmuss50** [comment]
I dont see this ever being done in loom, as like you said it will cause more issues than it solves. There should be nothing stopping your from doing it in a 3rd party plugin though.

**Comment by Eufranio** [comment]
This would indeed be a great feature. If anyone seeing this ever finds out or develop a 3rd party plugin to inject interfaces into non-mod classes, please let me know! :)
